kernel/scheduler.c

```diff
@@ -219,6 +222,8 @@ struct __attribute__((packed)) task_block {
 
     /* threads */
     struct task **threads;
+    struct task *joined_thread;
```

Comment:
I agree with you on changing the name of `joined_thead` to `joiner_thread`, it is semantically more correct.

About safety, the case you described can never happen because on thread creation the soon-to-be called `joiner_thread` field is set to NULL (scheduler.c:1382), therefore having a pointer to a `task struct` is completely safe and does not require doing a linear scan of the list of threads.